From 658260c3eec7e9edfc19574298adb86429fe5f66 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 21 Feb 2022 08:27:06 +0200
Subject: [PATCH] [SW-0] rom: revert sp sram if usage

This is a "revert me" patch.

Currently the dependency of hl_spsram_if.h on hl_boot_desc.h is
causing CI failures. Till a working solution is concluded,
this patch reverts the usage to unblock CI.

Change-Id: I00f4c76e0511379de3b1c4650e8c25b955a12395
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 6ff4043..80c9ba5 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -38,7 +38,6 @@
 		.max_imgs = HL_FW_IMG_SPI_MAX_BOOT_PTRS,
 		.load_addr = 0,
 	},
-	.sp_sram = (struct hl_spsram_layout *)(HL_GEN_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET),
 };
 
 #define ARCBOOT_STATUS_REG_SZ_BITS		32
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 7381e9d..c4bbf70 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -25,8 +25,9 @@
 #define ALIGN_VAL(val, byte)	ALIGN_SUB(val, (byte - 1))
 
 static struct hl_fw_boot_ptrs_ctx *p_img_ptr_ctx;
-static struct hl_fw_boot_ptrs *p_boot_ptr;
-static struct hl_fw_img_desc *p_img_desc;
+static struct hl_fw_boot_ptrs *p_boot_ptr = (struct hl_fw_boot_ptrs *)(HL_SPI_BOOT_PTR_SP_SRAM_ADDR);
+static struct hl_fw_img_desc *p_img_desc = (struct hl_fw_img_desc *)(HL_SPI_BOOT_PTR_SP_SRAM_ADDR +\
+					BOOT_PTRS_SZ);
 
 static bool boot_desc_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
@@ -47,10 +48,7 @@
 {
 	p_img_ptr_ctx = &arcboot_rom_global.boot_ptr_ctx[img];
 
-	if (img == HL_FW_BOOT_PTR_SPI) {
-		p_boot_ptr = &arcboot_rom_global.sp_sram->img_ptr.boot_ptr;
-		p_img_desc = arcboot_rom_global.sp_sram->img_ptr.img_desc;
-	} else { /* EROM img ptr */
+	if (img == HL_FW_BOOT_PTR_EROM) {
 		p_boot_ptr = &arcboot_rom_global.img_ptrs_erom.boot_ptr;
 		p_img_desc = arcboot_rom_global.img_ptrs_erom.img_desc;
 	}
diff --git a/include/arcboot.h b/include/arcboot.h
index 3aa600b..ca074ce 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -15,7 +15,6 @@
 #include "regmap.h"
 #include "regmask.h"
 #include "hl_boot_desc_rom.h"
-#include "hl_spsram_if.h"
 
 #ifndef __weak
 #define __weak		__attribute__((weak))
@@ -76,7 +75,6 @@
 	bool fuse_spi;
 	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
 	struct hl_img_ptrs_erom img_ptrs_erom;
-	struct hl_spsram_layout *sp_sram;
 };
 
 enum arcboot_rom_bringup_proggress {
